### PR TITLE
MULE-12609: Every exported Mule package must contain api

### PR DESCRIPTION
**Pull request by pablokraan**
_ Moving classes from org.mule.runtime.core.execution to api/internal
_ Moving classes from org.mule.runtime.core.notification to api/internal
_ Moving more classes from org.mule.runtime.core.management.stats to api/internal
_ Moving more classes from org.mule.runtime.core.config to api/internal

### CODE REVIEW

File: http/src/test/java/org/mule/test/module/http/functional/listener/HttpListenerLifecycleTestCase.java [hunk 1]
```diff
@@ -44,15 +45,15 @@ protected String getConfigFile() {
 
   @Test
   public void stopOneListenerDoesNotAffectAnother() throws Exception {
-    ExtensionMessageSource httpListener = (ExtensionMessageSource) ((Flow) getFlowConstruct("testPathFlow")).getSource();
+    ExtensionComponent httpListener = (ExtensionComponent) ((Flow) getFlowConstruct("testPathFlow")).getSource();
```
Comment (elrodro83) [inline http/src/test/java/org/mule/test/module/http/functional/listener/HttpListenerLifecycleTestCase.java]: why this change?

Comment (pablokraan) [inline http/src/test/java/org/mule/test/module/http/functional/listener/HttpListenerLifecycleTestCase.java]: This was the only place outside extension support modules referencing ExtensionMessageSource, but it is not required to reference it as the only used method exists on ExtensionComponent.
It could be debated why if ExtensionComponent is API then why ExtensionMessageSource is not... I prefer to expose the minimum number of classes, unless you disagree
